### ISSUE TITLE
Bots could use the Chronosphere

### ISSUE THREAD

**Issue by Mailaender** [OpenRA/OpenRA#18267]
This is probably difficult to implement.

**Comment by KOYK** [comment]
Maybe use AI's chrono as a 2 step thing as a work around. 
First the AI "tags" its units and then give the illusion of chrono by doing the chrono visual effect, reset timer and just move those units in the target location.

**Comment by dnqbob** [comment]
ummm...  There is a idea that won't change the engine and if you and players don't mind.

You can simply make an alternative support power only for bot owner which create several tanks at certain hostile position and disappear after count down,  and use the Chronosphere sound effect.

**Comment by dnqbob** [comment]
To apply an actually meaningful Chronosphere operation,  there is much things to fix to the engine,  which includes:
1.  Squad fix.  A chrono squad will get in formation for this support power. It is better to fix entire squad system like the planning that the AI project will do. 
2.  AI support power fix for Chronosphere only. 

I think the two above is probably enough to make related section refactored. So if you demand a quick and convenient application,  you can use the comment above.

**Comment by KOYK** [comment]
I had that idea too, tho the problem with it is that you will get tanks to spawn from no ware, that could be lame for some players the same way as killing the AIs ore trucks to stop its economy while the AI gets money every X seconds out of no ware as a bonus.
Hence I end up to this, like having the AI actually build the units and then just remove them from map and spawn new ones at the enemy location. I mean this can be a work around until the AI gets a major face lift. I am sure it can be done in LUA in user maps.

**Comment by dnqbob** [comment]
Yes, it is a problem. I applied my idea in SP, and it localization version which is released. To solve this I change this Bot-Only superweaon's name to "Instant Wormhole(AI)" and in description I tell players it is a cheating weapon by AI that will spawn units from nowhere.

I think RA can be better at here though, the units will get killed after a time (fake teleport back) which means they will only exist for a short time and will not join the AI rush team.

We always expain to our localization version players that AI cheat is a have to. AI needs to be tough to make players prepared for PVP, not in order to beat players out of sense.

**Comment by dnqbob** [comment]
But... what if player is going to hijack one of the fake units?

**Comment by KOYK** [comment]
> But... what if player is going to hijack one of the fake units?

Hmm yeah, also husks return after chrono, I don't remember if they return after an enemy mechanic repairs them or not, I guess they do return? but still if thats the case it feels hard to solve.

**Comment by KOYK** [comment]
The problem with the chronosphere power is that it needs to be "clicked" 2 times, once on the units and once on the target location. I don't know if it possible to fool the bots to consider the chrono like it is a two powers and have them use it twice with a delay. Tho I doubt that is possible no matter how optimistic I am for that idea :smile: , without adding some short of C# code I guess, tho I have no idea about C#, so you guys know better if thats a good idea or not.

At least IC feels more easy to add, to the AI, I guess by using some conditions based on units that are low on health or something. Tho thats an other subject.

**Comment by dnqbob** [comment]
Engine code needs a lot of improvements to finish this, and we also need to think if the code here can be reuse to other components in the future and make senses in logic for people who maintain this. So... a lot of work&thinking.

**Comment by dnqbob** [comment]
> The problem with the chronosphere power is that it needs to be "clicked" 2 times, once on the units and once on the target location. I don't know if it possible to fool the bots to consider the chrono like it is a two powers and have them use it twice with a delay. Tho I doubt that is possible no matter how optimistic I am for that idea 😄 , without adding some short of C# code I guess, tho I have no idea about C#, so you guys know better if thats a good idea or not.

Actually it is more difficult in application: not only 2 clicks, but also the first click needs to cast on actor location while the later click needs to be a cell (or cells) can be entered.

But anyway, I have finished [#22177](https://github.com/OpenRA/OpenRA/pull/22177). You can have a test.
